client/app/pages/queries/view.js

```diff
@@ -204,6 +204,25 @@ function QueryViewCtrl($scope, Events, $route, $routeParams, $location, $window,
     $scope.saveQuery(undefined, { is_draft: $scope.query.is_draft });
   };
 
+  $scope.saveAsDataset = () => {
```

Comment:
Unfortunately, [it didn't quite work](https://stackoverflow.com/questions/33527653/babel-6-regeneratorruntime-is-not-defined).